### ISSUE TITLE
Accessibility props for Icon

### ISSUE THREAD

**Issue by tfsjohan** [react-native-training/react-native-elements#1444]
**Is your feature request related to a problem? Please Describe.**
The Icon component can't expose accessibility labels and traits. For buttons this is a huge problem for accessability.

**Describe the solution you'd like**
Set the button trait if onPress is specified and expose the other accessibility props like label to help VoiceOver users know what the icon button does.

**Describe alternatives you've considered**
Create a custom component or button

**Comment by iRoachie** [comment]
Hey @tfsjohan, i'm don't know much about accessibility, can you propose a way we can add this in?

**Comment by tfsjohan** [comment]
The short answer would be to expose all the accessibility props that React Native supports on the Icon component. You can read more about these here:
https://facebook.github.io/react-native/docs/accessibility

The longer answer is that VoiceOver can't see the icon element so it is impossible to "tap" on an <Icon/> with a onPress handler because VoiceOver won't know it is an element. 

As a minimum I would like to have the accessibilityRole set default to "button" if there is an onPress handler. The accessible prop should be true in all cases. 

The accessibilityLabel is used to describe the icon so that VoiceOver reads what the button actually do instead of just "button" (which would be the default if accessibilityRole="button") or nothing if it's not marked as an accessible element. This is basically the "alt" / "title" attribute if you are familiar with html. 

The accessibilityHint is used to have a longer explanation of that the button does. 

These four attributes is what I really need, but maybe someone else would like all the accessibilityProps. But I do think these are the most important ones.

**Comment by tfsjohan** [comment]
I'm using TypeScript, so maybe it would be enough to update the typings to either allow any attribute or expose the accessibility props in the d.ts file. I noticed in the code that you set any attribute specified on the <Icon> component.

**Comment by tfsjohan** [comment]
<Button> has the exact same problems as <Icon> by the way.

**Comment by xavier-villelegier** [comment]
@tfsjohan Actually, everything you pass to the `Button` is passed to the touchable component, so you should be able to use the accessibility props https://github.com/react-native-training/react-native-elements/blob/1f76514f863ea48899236b35f5e963ce6ecb397f/src/buttons/Button.js#L71-L76

Same goes for the `Icon`
https://github.com/react-native-training/react-native-elements/blob/1f76514f863ea48899236b35f5e963ce6ecb397f/src/icons/Icon.js#L37-L38
We're using `TouchableHighlight` for the `Icon` do you know if the accessibility props work for it too or we need to switch to `TouchableOpacity` ?

**Comment by xavier-villelegier** [comment]
Looks like the accessibility props are compatible with any View, so you can use `Button` and `Icon` this way:

```jsx
<Button
  accessible
  accessibilityLabel="Tap me!"
  title="My button"
/>

<Icon
  accessible
  accessibilityLabel="Tap me!"
  name='sc-telegram'
  type='evilicon'
/>
```
